_bleio/common.py

```diff
@@ -327,15 +328,15 @@ def _start_scan_hcitool(
             )
         # pylint: enable=consider-using-with
         # Throw away the first two output lines of hcidump because they are version info.
-        hcidump.stdout.readline()
-        hcidump.stdout.readline()
+        hcidump.stdout.readline()  # type: ignore[union-attr]
+        hcidump.stdout.readline()  # type: ignore[union-attr]
         returncode = self._hcitool.poll()
         start_time = time.monotonic()
-        buffered = []
+        buffered: List[bytes] = []
```

Comment:
We can't do types on variables like this for now. 

The types for function arguments and returns are the only ones we can do ATM I believe. 

We tried a few of them like this early on when we started the typing effort and found that mpy-cross doesn't handle typed variables like this correctly, it does not get tripped up by argument or return types though so those are the ones we started working on .

Reply:
Sorry about that and thanks for the explanation! How hard would it be to add variable types to the grammar for circuitpython? I think in CPython it would be fairly easy (I happily just got a tour of the current tooling for updating the grammar - it's super streamlined at this point).

Reply:
> We tried a few of them like this early on when we started the typing effort and found that mpy-cross doesn't handle typed variables like this correctly, it does not get tripped up by argument or return types though so those are the ones we started working on .

I am not sure this is true any more. I just tried it in the REPL in 7.x and there was no complaint. I don't remember this not working recently.

Reply:
After poking around, my understanding is that this code is never run on circuitpython, right?

Reply:
Ah whoops I was in "circuitpython library" review mode when I looked at this initially. I know for use on Raspberry Pi and other SBC it does run in CPython and not circuitpython. 

The one use-case I am less familiar with is using Blinka libraries with micropython. I believe it can be used for that as well, but I don't know specifically about this ble library or whether they get mpy-crossed when used in that capacity.

Reply:
> I am not sure this is true any more. I just tried it in the REPL in 7.x and there was no complaint. I don't remember this not working recently.

if I recall correctly the plain python code ran fine on a device, but it raised an error when mpy-cross was run on it. I do think the mpy version might have been changed since back when I tried that before though so potentially it handles it now.

Reply:
Y'all have been super responsive. Totally makes sense you'd be in circuitpython mode given the sprint! Thank you!